src/cargo/core/resolver/context.rs

```diff
@@ -329,14 +330,16 @@ impl Context {
     }
 
     pub fn graph(&self) -> Graph<PackageId, Vec<Dependency>> {
-        let mut graph = Graph::new();
+        let mut graph: Graph<PackageId, Vec<Dependency>> = Graph::new();
         self.activations
             .values()
             .for_each(|(r, _)| graph.add(r.package_id()));
         for i in self.parents.iter() {
             graph.add(*i);
             for (o, e) in self.parents.edges(i) {
-                *graph.link(*o, *i) = e.to_vec();
+                let old_link = graph.link(*o, *i);
+                debug_assert!(old_link.is_empty());
```

Comment:
Oh FWIW I'd be fine with just a plain `assert!`, I think we should only use `debug_assert` if there's a runtime issue